agg-svn/agg-2.4/include/agg_rasterizer_cells_aa.h

```diff
@@ -176,14 +178,16 @@ namespace agg
     }
 
     //------------------------------------------------------------------------
-    template<class Cell> 
+    template<class Cell>
     AGG_INLINE void rasterizer_cells_aa<Cell>::add_curr_cell()
     {
         if(m_curr_cell.area | m_curr_cell.cover)
         {
             if((m_num_cells & cell_block_mask) == 0)
             {
-                if(m_num_blocks >= cell_block_limit) return;
+                if(m_num_blocks >= m_cell_block_limit) {
+                    throw std::overflow_error("Exceeded cell block limit");
```

Comment:
This is _literally_ the only exception being thrown by the included Agg code. I'm not sure how I feel about that.